tests/run-make/const-prop-lint/rmake.rs

```diff
@@ -0,0 +1,19 @@
+// Test that emitting an error because of arithmetic overflow lint does not leave .o files around
+// because of interrupted codegen.
+
+use std::fs;
+
+use run_make_support::{rustc, tmp_dir};
+
+fn main() {
+    rustc().input("input.rs").run_fail_assert_exit_code(1);
+
+    for entry in fs::read_dir(tmp_dir()).unwrap() {
```

Comment:
From what I read in `run-make-support`, `tmp_dir` is added as `current_env` for commands like `rustc()`. So yes. :)

Reply:
Cool, thank you 👍

Reply:
> From what I read in `run-make-support`, `tmp_dir` is added as `current_env` for commands like `rustc()`. So yes. :)

To be pedantic: the `rustc()` helper constructs a `Rustc` command wrapper which by default has `--out-dir` set to `tmp_dir()` because that's what most run-make tests seem to want.

Reply:
If we *want* to be pedantic, then `tmp_dir` could have a name that showcases its semantics better: it's not that it's temporary that is important, but that it is unique per-test, and that all the root wrappers use it as their out dir.

Reply:
Yes, I should think about giving that a better name, and better document it.

Reply:
I opened https://github.com/rust-lang/rust/issues/125726 to track this.